tock-process-console/src/ui_management/pages/setup_page/setup_page.rs

```diff
@@ -33,21 +48,48 @@ impl From<&State> for Properties {
     }
 }
 
+struct EventHandler {}
+
 /// Struct that handles setup of the console application
 pub struct SetupPage {
     input_box: InputBox,
     action_sender: UnboundedSender<Action>,
     properties: Properties,
+    scrollbar_state_serial: ListState,
+    scrollbar_state_boards: ListState,
+    probeinfo_sender: Sender<Vec<String>>,
+    probeinfo_receiver: Receiver<Vec<String>>,
+    showed_serials: bool,
+    showed_boards: bool,
 }
 
 impl SetupPage {
     fn set_port(&mut self) {
-        // should update the port
-        if self.input_box.is_empty() {
-            return;
+        let probeinfo = match self.probeinfo_receiver.try_recv() {
+            Ok(probeinfo) => probeinfo,
+            Err(error) => panic!("{}", error),
+        };
+
+        if probeinfo.is_empty() {
+            print!("No Boards Found!")
         }
 
-        let port = self.input_box.text();
+        let mut port_number = 0;
+
+        if self.showed_boards == true {
+            port_number = match self.scrollbar_state_boards.selected() {
+                Some(num) => num,
+                None => panic!("Error selecting the board!"),
+            };
+        }
+        if self.showed_serials == true {
+            port_number = match self.scrollbar_state_serial.selected() {
+                Some(num) => num,
+                None => panic!("Error selecting the board!"),
+            };
+        }
+
+        let port = probeinfo[port_number].clone();
         let _ = self.action_sender.send(Action::ConnectToBoard {
             port: port.to_string(),
```

Comment:
No need for .to_string(). Also, handle the error case
```suggestion
let x = self.action_sender.send(Action::ConnectToBoard { port });
```